### PR TITLE
Test suite doc; tweaks

### PR DESCRIPTION
**Pull request by chrisarcand**
I've tweaked the server startup doc to be a bit more clear and added a doc page on running the test suites.

Suggestions welcome, especially for the Self Service setup as I just made that up quickly without looking too closely at the runner.

### CODE REVIEW

**Comment by chrisarcand** [comment]
cc/ @Fryguy

File: developer_setup/running_test_suites.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+* **javascript**
+* **self_service**
+* **brakeman**
+
+Each suite has a pair of Rake tasks to setup and run them, in the form of `test:<type>` and
+`test:<type>:setup` (if required):
+
+```bash
+rake test:vmdb                   # Run all specs except migrations, replication, and automation
+rake test:vmdb:setup             # Setup environment for vmdb specs
```
Comment (kbrock) [inline developer_setup/running_test_suites.md]: would it make more sense to list the setup before the running step?
(although, I guess alphabetical makes sense to me too)

File: developer_setup.md [hunk 2]
```diff
@@ -155,9 +155,12 @@ git fetch other_user
 ### Get the Rails environment up and running
 
 ```bash
-bin/setup
+bin/setup                  # Installs dependencies, config, prepares database, etc
+bundle exec rake evm:start # Starts the ManageIQ EVM Application in the background
```
Comment (kbrock) [inline developer_setup.md]: do we need to run `evm:start` and rails s?

May want more of a note around configuring fewer resources / spartian stuff here.

File: developer_setup/running_test_suites.md (hunk 1 above)
Comment (chrisarcand) [inline developer_setup/running_test_suites.md]: I just pasted in what Rake fed me, but did go out of my way to put the main test suite up top. Sure, makes sense to me :)

File: developer_setup.md (hunk 2 above)
Comment (chrisarcand) [inline developer_setup.md]: Hmm I'm not sure what you mean here; haven't dug in enough to know about those options or what's available. Is this beyond the minimal mode guide referenced below?
I'll add whatever specific things you think would be helpful.

**Comment by chrisarcand** [comment]
Rebased and ready for another look.

**Pull request merged**